### ISSUE TITLE
should the markdown package be in "Imports" rather a "Suggests" in DESCRIPTION?

### ISSUE THREAD

**Issue by rharlow86** [rstudio/htmltools#127]
the exported function includeMarkdown from htmltools makes a call to markdown::markdownToHTML, does that mean that it should be an Import rather than a Suggests?

If so, I am happy to submit a PR.

**Comment by yihui** [comment]
No. Technically (i.e., for `R CMD check`) it is fine for **markdown** to be in `Suggests`. `includeMarkdown` is not an essential feature of **htmltools**, and none of other functions need the **markdown** package. In general, we want to keep the dependencies as light as possible.

**Comment by rharlow86** [comment]
I am all for keeping the dependencies as light as possible, but an end user of the package who tries to use the exported function will throw an error until they install markdown, which makes markdown feel like a dependency to me, but I can understand how there is some subjectivity to “essential.” Is that the general rule of thumb? (E.g. If another package is used only once, use suggests?)

**Comment by yihui** [comment]
I wouldn't say it is a general rule of thumb, but there is certainly some subjectivity. The trade-off could be different to different authors---whether you want perfect user experience or light dependencies (avoid forcing users to install a package that they rarely need). In this particular case, I wouldn't make **markdown** a hard dependency of **htmltools**. Another reason is that **markdown** is a hard dependency of several other (popular) packages like **knitr**, so I believe this should rarely be a problem.
